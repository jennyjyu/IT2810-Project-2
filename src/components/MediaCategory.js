import React, {Component} from 'react'



class MediaCategory extends Component {
    constructor(props) {
        super(props)
        this.state = {
            categories: props.categories,
        }
    }

    isOptionFavorite(i) {
        if (this.props.categories[i].text === this.props.favoriteCombo.favoriteText) {
            return true;
        }
        if (this.props.categories.text === this.props.favoriteCombo.favoriteImage) {
            return true;
        }
        if (this.props.categories.text === this.props.favoriteCombo.favoriteSound) {
            return true;
        }
        return false;
    }
    // renderCheckbox kan utvides til å ta input for å lage en liste kategorier med props
    renderCheckbox(i) {
        return (
            <label className="radio-label" key={Math.random()}>
                <input
                    id={this.props.id}
                    type="radio"
                    checked={i === this.props.index }
                    defaultChecked={this.props.categories[i].checked}
                    name={this.props.mediaLabel}
                    onClick={() => this.props.handleRadioChange(this.props, i)}>
                </input>
                {this.props.categories[i].text}
                <br/>
            </label>
        )
    }


    createCheckbox = () => {
        let checkboxes = []
        if (this.state.categories){ // This ensures that state.categories is set before for loop. prevents undefined .length below
            for (let i=0; i < this.state.categories.length; i++) {
                checkboxes.push(this.renderCheckbox(i))
            }
        }
        return checkboxes
    }

    render() {

        return(
            <div className="category-container">
                <label>
                    {this.props.mediaLabel}
                <ul className="media-list">
                    {this.createCheckbox()}
                </ul>
                </label>

            </div>
        )
    }
}
export default MediaCategory
