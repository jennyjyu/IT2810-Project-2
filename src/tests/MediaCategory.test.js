import React from "react"
import MediaCategory from "../components/MediaCategory"
import renderer from "react-test-renderer"
// import App from "../App"



test("MediaCategory snapshot test", () => {
  const component = renderer.create(
    <MediaCategory />
  )
  let tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})
