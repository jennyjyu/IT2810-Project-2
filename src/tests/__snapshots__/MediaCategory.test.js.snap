// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaCategory snapshot test 1`] = `
<div
  className="category-container"
>
  <label>
    <ul
      className="media-list"
    />
  </label>
</div>
`;
